src/model/utils/deletecontent.js

```diff
@@ -63,6 +63,13 @@ export default function deleteContent( model, selection, options = {} ) {
 	const schema = model.schema;
 
 	model.change( writer => {
+		const selRange = selection.getFirstRange();
+
+		// If the selection is already removed, don't do anything.
+		if ( selRange.root.rootName == '$graveyard' ) {
```

Comment:
Why not outside, right after checking `selection.isCollapsed`? It seems more natural to have there all the early returns. Since you've put it here, in the `model.change()` block I had to scroll the code down to see if there's anything after `model.change()` that might need (or perhaps should not) be executed.

Reply:
I did it like this to be sure that the selection doesn't get into the graveyard before `model.change` actually executes, but I forgot that it executes right away in all circumstances (as opposed to `enqueueChange`). Will fix.